### PR TITLE
chore(apigatewayv2): integration api re-organization

### PR DESCRIPTION
**Pull request by nija-at**
There are three major changes.

`HttpRouteIntegration` (and its sibling `WebSocketRouteIntegration`)
creates a CDK construct (`HttpIntegration` and `WebSocketIntegration`)
as part of its bind operation. The id to this CDK construct is
determined by hashing the results of the bind.
Using hashes makes the construct id fragile/sensitive, consequently the
CFN resource's logical id fragile.
The fragility comes mainly from the question - have we hashed all of the
expected properties that should be hashed, and nothing extra?
If we have not hashed properties that should be there, or hashed too
much, we end up with a hash change, hence resource replacement that
is unexpected.

This commit changes this approach and asks the user to provide the
construct's id. This is more aligned with the current CDK expectation
that users provide an id when initializing constructs.
We just don't have a good way to validate that our hashing is accurate,
so let's not do it at all. This change makes the user provide a unique
name within a scope, which is already a standard requirement for CDK
constructs.

Secondly, the ergonomics of specific integration implementations, such
as, `LambdaProxyIntegration`, `HttpAlbIntegration`, etc. is modified so
that the integrating primitive is moved out of the 'props', and to the
constructor.
The API ergonomics of this feels much better than having to always
provide a 'props'.

Since this package contains constructs around both http api and
websocket api, the convention to follow is that all classes specific to
the former will be prefixed with `Http` and the latter will be prefixed
with `WebSocket`.
Bring the integration classes `LambdaProxyIntegration` and
`HttpProxyIntegration` in line with this convention. These are renamed
to `HttpLambdaIntegration` and `HttpUrlIntegration` respectively.

BREAKING CHANGE: The `HttpIntegration` and `WebSocketIntegration`
classes require an "id" parameter to be provided during its initialization.
* **apigatewayv2-integrations:** The `LambdaWebSocketIntegration` is now
  renamed to `WebSocketLambdaIntegration`. The new class accepts the
  handler to the target lambda function directly in its constructor.
* **apigatewayv2-integrations:** `HttpProxyIntegration` and
  `HttpProxyIntegrationProps` are now renamed to `HttpUrlIntegration`
  and `HttpUrlIntegrationProps` respectively. The new class accepts the
  target url directly in its constructor.
* **apigatewayv2-integrations:** `LambdaProxyIntegration` and
  `LambdaProxyIntegrationProps` are now renamed to
  `HttpLambdaIntegration` and `HttpLambdaIntegrationProps` respectively.
  The new class accepts the lambda function handler directly in its
  constructor.
* **apigatewayv2-integrations:** `HttpAlbIntegration` now accepts the
  ELB listener directly in its constructor.
* **apigatewayv2-integrations:** `HttpNlbIntegration` now accepts the
  ELB listener directly in its constructor.
* **apigatewayv2-integrations:** `HttpServiceDiscoveryIntegration` now
  accepts the service discovery Service directly in its constructor.
* **apigatewayv2-authorizers:** `UserPoolAuthorizerProps` is now
  renamed to `HttpUserPoolAuthorizerProps`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nija-at** [comment]
cc @ayush987goyal

**Review by otaviomacedo — APPROVED** [review]

**Comment by nija-at** [comment]
@ayush987goyal - thanks for your comments.

> Like for the HTTP Lambda integration, the docs uses proxy everywhere ([ref](https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html)). Similarly for HTTP Proxy ([ref](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-as-simple-proxy-for-http.html))

The links above point to REST API (v1) and not HTTP API (v2).
In the [v2 docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html), this is still present but its not stressed as much. 

Nevertheless, I don't see the value of keeping the word "proxy" in the name for either of these cases. I don't see any other kind. Maybe I'm missing something?

**Comment by ayush987goyal** [comment]
> Nevertheless, I don't see the value of keeping the word "proxy" in the name for either of these cases. I don't see any other kind. Maybe I'm missing something?

The only value that I was seeing was to have a parity with the docs. Not sure if that is even a concern to be honest or something we try to ensure in L2/3 constructs. We might anyway be pointing to the docs in the README and that should suffice.

File: packages/@aws-cdk/aws-apigatewayv2-authorizers/README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ The example below showcases default authorization, along with route authorizatio
```
[the remark this answers is not in the document]
Comment (nija-at) [reply packages/@aws-cdk/aws-apigatewayv2-authorizers/README.md:71]: Updated in the next rev.

File: packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/alb.ts [hunk 2]
```diff
@@ -8,24 +8,24 @@ import { HttpPrivateIntegration } from './private/integration';
...
 
 /**
  * The Application Load Balancer integration resource for HTTP API
  */
 export class HttpAlbIntegration extends HttpPrivateIntegration {
-  constructor(private readonly props: HttpAlbIntegrationProps) {
-    super();
+  constructor(
+    id: string,
+    private readonly listener: elbv2.IApplicationListener,
```
[the remark this answers is not in the document]
Comment (nija-at) [reply packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/alb.ts:19]: Good catch. Fixed.

File: packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/http-proxy.ts [hunk 3]
```diff
@@ -34,17 +29,22 @@ export interface HttpProxyIntegrationProps {
 /**
  * The HTTP Proxy integration resource for HTTP API
  */
-export class HttpProxyIntegration extends HttpRouteIntegration {
-  constructor(private readonly props: HttpProxyIntegrationProps) {
-    super();
+export class HttpUrlIntegration extends HttpRouteIntegration {
+  /**
+   * @param id id of the underlying integration construct
+   * @param listener the URL to proxy to
```
Comment (ayush987goyal) [inline packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/http-proxy.ts:35]:
```suggestion
   * @param url the URL to proxy to
```

File: packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/service-discovery.ts [hunk 4]
```diff
@@ -7,18 +7,23 @@ import { HttpPrivateIntegration } from './private/integration';
...
 
 /**
  * The Service Discovery integration resource for HTTP API
  */
 export class HttpServiceDiscoveryIntegration extends HttpPrivateIntegration {
-  constructor(private readonly props: HttpServiceDiscoveryIntegrationProps) {
-    super();
+  /**
+   * @param id id of the underlying integration construct
+   * @param listener the service discovery resource to integrate with
```
Comment (ayush987goyal) [inline packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/service-discovery.ts:18]:
```suggestion
   * @param service the service discovery resource to integrate with
```

File: packages/@aws-cdk/aws-apigatewayv2-integrations/lib/websocket/lambda.ts [hunk 5]
```diff
@@ -6,29 +6,28 @@ import {
...
  */
-export interface LambdaWebSocketIntegrationProps {
+export class WebSocketLambdaIntegration extends WebSocketRouteIntegration {
+
+  private readonly _id: string;
+
   /**
-   * The handler for this integration.
+   * @param id id of the underlying integration construct
+   * @param listener the Lambda function handler
```
Comment (ayush987goyal) [inline packages/@aws-cdk/aws-apigatewayv2-integrations/lib/websocket/lambda.ts:20]:
```suggestion
   * @param handler the Lambda function handler
```

File: packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/http-proxy.ts (hunk 3 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/http-proxy.ts:35]: Well sheeet 🤦‍♂️

Thanks!

**Review by ayush987goyal — APPROVED** [review]

**Pull request merged**